pkg/api/customization/authn/user_actions.go

```diff
@@ -28,7 +31,13 @@ type Handler struct {
 func (h *Handler) Actions(actionName string, action *types.Action, apiContext *types.APIContext) error {
 	switch actionName {
 	case "changepassword":
-		return h.changePassword(actionName, action, apiContext)
+		err := h.changePassword(actionName, action, apiContext)
+		if err == nil {
+			if strings.EqualFold(settings.FirstLogin.Get(), "true") {
+				settings.FirstLogin.Set("false")
+			}
+		}
+		return err
 	case "setpassword":
 		return h.setPassword(actionName, action, apiContext)
```

Comment:
The setpassword action and PUTting to an authConfig should clear the setting too.  Automated scripts to setup an install might do any of the three